### PR TITLE
1 list of crypto currencies

### PR DESCRIPTION
**Pull request by rodrigomp88**

### CODE REVIEW

File: src/components/Button/RegularButton.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React, {FunctionComponent} from 'react';
+import {ButtonProps} from '../../interfaces';
+import {colors} from '../colors';
+import RegularText from '../Texts/RegularText';
+import {ButtonView} from './styles';
+
+const RegularButton: FunctionComponent<ButtonProps> = props => {
```
Comment (tomasdematos) [inline src/components/Button/RegularButton.tsx]: use destructuring in the props, and implicit return

File: src/components/Crypto/CryptoItem.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  LeftData,
+  LeftView,
+  RightView,
+} from './styles';
+
+import up from '../../assets/arrow-up.png';
+import down from '../../assets/arrow-down.png';
+
+const CryptoItem: FunctionComponent<CryptoProps> = props => {
+  return (
```
Comment (tomasdematos) [inline src/components/Crypto/CryptoItem.tsx]: use implicit return

File: src/components/Crypto/CryptoItem.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  CryptoItemRow,
+  LeftData,
+  LeftView,
+  RightView,
+} from './styles';
+
+import up from '../../assets/arrow-up.png';
+import down from '../../assets/arrow-down.png';
+
+const CryptoItem: FunctionComponent<CryptoProps> = props => {
```
Comment (tomasdematos) [inline src/components/Crypto/CryptoItem.tsx]: use destructuring

File: src/components/Crypto/CryptoLogo.tsx [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import React, {FunctionComponent} from 'react';
+import {CryptoLogoProps} from '../../interfaces';
+import {Image, StyledView} from './styles';
+
+const CryptoLogo: FunctionComponent<CryptoLogoProps> = props => {
+  return (
```
Comment (tomasdematos) [inline src/components/Crypto/CryptoLogo.tsx]: implicit return and destructuring

File: src/components/Crypto/CryptoSection.tsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+import CryptoItem from './CryptoItem';
+import {CryptoList, CryptoSectionBackground} from './styles';
+
+const CryptoSection: FunctionComponent<CryptopSectionProps> = props => {
+  return (
+    <CryptoSectionBackground>
+      <CryptoList
+        data={props.data}
+        showsVerticalScrollIndicator={false}
+        keyExtractor={({id}: any) => id.toString()}
```
Comment (tomasdematos) [inline src/components/Crypto/CryptoSection.tsx]: don't use type any, also flatlist, will send the type to the function already

File: src/components/Crypto/CryptoSection.tsx [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+        renderItem={({item}: any) => <CryptoItem {...item} />}
```
Comment (tomasdematos) [inline src/components/Crypto/CryptoSection.tsx]: move to a render function, don't use any

File: src/components/Crypto/styles.tsx [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+  width: 48px;
+  height: 48px;
+  border-radius: 50px;
+  resize-mode: cover;
+`;
+
+//Old
+export const Container = styled.View`
+  height: 80px;
+  width: 327px;
```
Comment (tomasdematos) [inline src/components/Crypto/styles.tsx]: you shoud use percentages if is too big

File: src/components/Crypto/styles.tsx [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+  left: 80%;
+`;
+
+export const Arrow = styled.View`
+  position: absolute;
+  left: 81%;
+  top: 55%;
+`;
+
+export const ArrowPorentage = styled.Image`
```
Comment (tomasdematos) [inline src/components/Crypto/styles.tsx]: Percentage

File: src/components/Texts/TitleText.tsx [hunk 9]
```diff
@@ -0,0 +1,11 @@
+import React, {FunctionComponent} from 'react';
+import {TextProps} from '../../interfaces';
+import {StyledTitleText} from './styles';
+
+const TitleText: FunctionComponent<TextProps> = props => {
+  return (
```
Comment (tomasdematos) [inline src/components/Texts/TitleText.tsx]: check all the implicit return if it don't have nothing before the return, and the destructuring of the props

File: src/components/container.tsx [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+import styled from 'styled-components/native';
+import {Dimensions} from 'react-native';
+
+export const Container = styled.View`
+  display: flex;
+  align-items: center;
+  background-color: ${colors.white};
+`;
+
+export const ScreenWidth = Dimensions.get('screen').width;
```
Comment (tomasdematos) [inline src/components/container.tsx]: this shoud be in the theme, like the colors: 

utils/theme.ts

const theme = {
   colors: ....
   screenWidth....
}

export default theme

File: src/interfaces/index.ts [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+import {
+  GestureResponderEvent,
+  ImageSourcePropType,
+  ImageStyle,
+  StyleProp,
+  TextStyle,
+  ViewStyle,
+} from 'react-native';
+
+export interface TextProps {
```
Comment (tomasdematos) [inline src/interfaces/index.ts]: declare the props in the componente like:

interface Props ....

File: src/interfaces/index.ts [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+  price: number;
+  percentage_last_24: number;
+}
+
+export interface CryptopSectionProps {
+  data: Array<CryptoProps>;
+}
+
+export interface CryptoLogoProps {
+  icon: any;
```
Comment (tomasdematos) [inline src/interfaces/index.ts]: don't use any

File: src/interfaces/index.ts [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+export interface ButtonProps {
+  btnStyles?: any;
```
Comment (tomasdematos) [inline src/interfaces/index.ts]: don't use any

File: src/navigator/RootStack.tsx [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+            paddingLeft: 10,
+          },
+          headerRight: () => (
+            <Profile
+              img={profile}
+              imgContainerStyle={{backgroundColor: colors.primary}}
+            />
+          ),
+        }}
+        // initialRouteName="Home"
```
Comment (tomasdematos) [inline src/navigator/RootStack.tsx]: delete comment

File: src/pages/Home/index.tsx [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+import {colors} from '../../components/colors';
+
+import CryptoSection from '../../components/Crypto/CryptoSection';
+import DataCrypto from '../../store/DataCrypto';
+import RegularButton from '../../components/Button/RegularButton';
+
+const Home: FunctionComponent = () => {
+  return (
+    <HomeContainer>
+      <StatusBar barStyle={'light-content'} backgroundColor={colors.primary} />
```
Comment (tomasdematos) [inline src/pages/Home/index.tsx]: barStyle="light-content"

File: styledComponents.d.ts [hunk 16]
```diff
@@ -0,0 +1 @@
+declare module 'styled-components/native';
```
Comment (tomasdematos) [inline styledComponents.d.ts]: unnecessary, already exist a lib for the types of styled, the error it throws when you use it should told you wich lib is

File: src/interfaces/index.ts (hunk 11 above)
Comment (rodrigomp88) [reply src/interfaces/index.ts]: all interface inside the file or only TextProps

**Comment by rodrigomp88** [comment]
Pass review

**Pull request merged**